pkg/core/naming/contextual.go

```diff
@@ -0,0 +1,43 @@
+package naming
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	core_registry "github.com/kumahq/kuma/pkg/core/resources/registry"
+)
+
+type sectionName interface {
+	~string | ~uint32
+}
+
+// MustContextualInboundName is a helper for code paths where you are certain the resource's
+// type exists in the global registry. Use it only when the resource and its type are guaranteed
+// to be valid.
+func MustContextualInboundName[T sectionName](r core_model.Resource, sectionName T) string {
+	name, err := ContextualInboundName(r, sectionName)
+	if err != nil {
+		panic(err)
+	}
+	return name
+}
+
+func ContextualInboundName[T sectionName](r core_model.Resource, sectionName T) (string, error) {
+	if r == nil {
+		return "", errors.New("cannot build contextual inbound name: resource is nil")
+	}
+
+	desc, err := core_registry.Global().DescriptorFor(r.Descriptor().Name)
+	if err != nil {
+		return "", errors.Wrapf(
```

Comment:
Is it any different than what was mentioned in error wrapping madr? https://github.com/kumahq/kuma/blob/3220f15dad5ecf795c5ff27321e4885830c66bf6/docs/madr/decisions/073-error-wrapping.md#L118

Reply:
ack